include/nbl/system/IFile.h

```diff
@@ -43,6 +43,9 @@ class IFile : public core::IReferenceCounted
 		return (m_flags & ECF_COHERENT) == ECF_COHERENT;
 	}
 
+protected:
+	virtual int32_t read(void* buffer, uint32_t offset, uint32_t sizeToRead) = 0;
+	virtual int32_t write(const void* buffer, uint32_t offset, uint32_t sizeToWrite) = 0;
```

Comment:
size_t instead of uint32_t

Reply:
yeah you dont want to fail on files > 4GB

Reply:
@sadiuk  changed?